azure/components/baseline/setup.ftl

```diff
@@ -260,9 +261,15 @@
                       "\"" + accountObject.Id + "\" || return $?",
                   "  #",
                   "  # Upload to keyvault if required.",
-                  "  if [[ ! $(az_check_secret" + " " +
+                  "  AZ_CHK_SECRET=$(az_check_secret" + " " +
                       "\"" + vmKeyVaultName + "\" " +
-                      "\"" + vmKeyPairName + "PublicKey" + "\") " +
+                      "\"" + vmKeyPairName + "PublicKey" + "\")",
+                  "  echo $\{AZ_CHK_SECRET}",
+                  "  if [[ $\{AZ_CHK_SECRET} " +
+                      "=~ \"does not have secrets get permission on key vault\" ]]; then",
+                  "    fatal \"The deployment user is not a member of the specified keyVault admin group\"",
+                  "  else",
```

Comment:
Might be better to break these into two if statements 
so the fatal section can exit with a non 0 return code to stop any futher scripts running 

Then you don't need a nested if statement. 

If that doesn't work the second if should be indented rather than inline with the initial if

Reply:
The other alternative is elif, which would be ok to use in this case as its a block related to a single task

Reply:
If you aren't a member of the admin group it doesn't make sense to try and do the private key having failed on the public key. I've added the indenting

Reply:
If that configuration isn't setup properly is there point in continuing with the rest of the script? 
the `fatal` function is a fatal log message and doesn't produce and non 0 exit code, so based on the current script it would log a fatal error but complete successfully. 

So it might be better to handle any fatal errors as indivdual steps and exit with a non 0 exit code then process the happy results once we have got past the requirements